src/processors/encounter/encounter-processor-helper.ts

```diff
@@ -300,6 +304,43 @@ export async function hydrateRepeatField(
         }),
     );
   }
+
+  const unMappedDiagnoses = encounter.diagnoses.filter((diagnosis) => {
+    return (
+      !assignedDiagnosesIds.includes(diagnosis?.diagnosis?.coded.uuid) &&
+      diagnosis.formFieldPath.startsWith(`rfe-forms-${field.id}_`)
+    );
+  });
+
+  const sortedDiagnoses = unMappedDiagnoses
+    .filter((diagnosis) => !diagnosis.voided)
+    .sort((a, b) => {
+      // Extract numeric part of formFieldPath for sorting
+      const numberA = parseInt(a.formFieldPath.split('_')[1], 10);
+      const numberB = parseInt(b.formFieldPath.split('_')[1], 10);
+      return numberA - numberB; // Sort numerically based on formFieldPath
+    });
+
+  if (field.type === 'diagnosis') {
```

Comment:
This might not specific to this PR but shouldn't we add an option to view the diagnosis in the Visit > encounters view. 
How does the form look in read-only mode?

Reply:
Let me do a screenshot

Reply:
> This might not specific to this PR but shouldn't we add an option to view the diagnosis in the Visit > encounters view. How does the form look in read-only mode?

They will be viewable there

Reply:
>They will be viewable there

The screen recording you shared doesn't show it. Maybe it's an old one

Reply:
> > They will be viewable there
> 
> The screen recording you shared doesn't show it. Maybe it's an old one

Its on the video, if you check the diagnosis on the visits widget and compare it to the open form they are the same, minute 1:32

Reply:
I think i should have been more clear, i meant the value for the diagnosis but i've also realized it's a similar behavior  as the orders. There's a requirement to show the entered values (i.e Infection due to Entamoeba, Mucus in stool, etc) in the visits view but that's out of scope for this PR

<img width="795" alt="Screenshot 2024-11-26 at 19 19 42" src="https://github.com/user-attachments/assets/cd8e769a-0101-454d-9066-cbcb58851a14">

Reply:
Yea, but i have updated the video to show the diagnosis well, however, on the visits widget, voided diagnosis are also displayed